### ISSUE TITLE
Residual Image

### ISSUE THREAD

**Issue by iancze** [MPoL-dev/MPoL#6]
**Is your feature request related to a problem or opportunity? Please describe.**

During the training loop, it would be useful to track the residual visibility plane and corresponding dirty image. A necessary but not sufficient condition for RML imaging success is that the residual image is dominated by thermal noise, i.e., the model has explained the data to within the measurement uncertainty. 

**Describe the solution you'd like**

A `ResidualTracker` or `GriddedResiduals` that can operate with the `FourierLayer.vis` and `Gridded.Dataset.vis` (note, not indexed, but full arrays) and compute residual Fourier products (Re, Im, Amp, Pha) and Images via the [inverse FFT](https://github.com/MPoL-dev/MPoL/blob/main/src/mpol/gridding.py#L291). Can also scale/display images with appropriate stretches {log(1 + |x|)} for ultimately feeding to Tensorboard. This needs to store actual tensors (not numpy arrays). 

It's unclear where this should live: images, datasets, utils, or a new module entirely?

Also requires API documentation and incorporation as a tutorial.

**Describe alternatives you've considered**

Operating with the preindexed visibilities would require an additional regridding/reindexing operation back to the full array, so I would think it would be better to intercept the products and recalculate the residuals while the arrays are still whole, though in principle the regridding is as simple as a masked assign.

This tracker will only be valid for the GriddedDataset. When we incorporate KaiserBessel interpolation, we will need to regrid the loose visibilities (analogous to CLEAN), though perhaps TorchKBNufft has this functionality.
